### PR TITLE
TST: np.argsort comparision

### LINKED ISSUE #17046: TST: tests comparison on numpy master
https://travis-ci.org/pandas-dev/pandas/jobs/256013691

this was working a few builds ago. I think maybe some random sorting details, IOW how exactly argsort works can make this fail with either '>' or '<'. so ok with changing test to accept either.

```
________________________ TestMixedIntIndex.test_argsort ________________________
[gw1] linux -- Python 3.6.2 /home/travis/miniconda3/envs/pandas/bin/python
self = <pandas.tests.indexes.test_base.TestMixedIntIndex object at 0x7fe54ca566a0>
    def test_argsort(self):
        idx = self.create_index()
        if PY36:
            with tm.assert_raises_regex(TypeError, "'>' not supported"):
>               result = idx.argsort()
pandas/tests/indexes/test_base.py:1850: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/util/testing.py:2493: in __exit__
    return self.exception_matches(exc_type, exc_value, trace_back)
pandas/util/testing.py:2528: in exception_matches
    raise_with_traceback(e, trace_back)
pandas/compat/__init__.py:343: in raise_with_traceback
    raise exc.with_traceback(traceback)
pandas/tests/indexes/test_base.py:1850: in test_argsort
    result = idx.argsort()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = Index([0, 'a', 1, 'b', 2, 'c'], dtype='object'), args = (), kwargs = {}
result = array([0, 'a', 1, 'b', 2, 'c'], dtype=object)
    def argsort(self, *args, **kwargs):
        """
            Returns the indices that would sort the index and its
            underlying data.
    
            Returns
            -------
            argsorted : numpy array
    
            See also
            --------
            numpy.ndarray.argsort
            """
        result = self.asi8
        if result is None:
            result = np.array(self)
>       return result.argsort(*args, **kwargs)
E       AssertionError: "'>' not supported" does not match "'<' not supported between instances of 'str' and 'int'"
pandas/core/indexes/base.py:2102: AssertionError
```

### PR DESCRIPTION
**Pull request by ysau**
- [ ] closes #17046
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master -u -- "*.py" | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -259,6 +259,7 @@ Indexing
 - Fixes bug where indexing with ``np.inf`` caused an ``OverflowError`` to be raised (:issue:`16957`)
 - Bug in reindexing on an empty ``CategoricalIndex`` (:issue:`16770`)
 - Fixes ``DataFrame.loc`` for setting with alignment and tz-aware ``DatetimeIndex`` (:issue:`16889`)
+- Fixes ``test_argsort``, ``test_numpy_argsort`` to cover both `>` and `<` in the error message (:issue:`17046`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: this doesn't need a release note

**Comment by jreback** [comment]
looks good. let's see if this passes. ping on green.

**Review by jreback — CHANGES_REQUESTED** [review]
comment

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (ysau) [inline doc/source/whatsnew/v0.21.0.txt]: Ok.  What do I need to do to undo changes in the release note?  Should I check in the code again and resubmit a pull request?

Comment (gfyoung) [inline doc/source/whatsnew/v0.21.0.txt]: On your branch at the top directory, perform the following:

~~~
git checkout master doc/source/whatsnew/v0.21.0.txt
~~~

Commit these changes and push.

**Review by gfyoung — APPROVED** [review]

**Comment by gfyoung** [comment]
@ysau : LGTM, ping when green!

**Comment by gfyoung** [comment]
Thanks @ysau !

**Comment by ysau** [comment]
@jreback, @gfyoung Thanks for the guidance!

**Pull request merged**